nothing sent to stackdriver - how to debug?

Sorry if this is more of a question than a bug report.

I have two very similar staging and production deployments running in GCE and I could only get zipkin to successfully forward traces to stackdriver in one of them.

I could only get the stackdriver forwarding to work in production for a few requests, then it stopped working on reboot for unknown magical reasons.

I am using the docker image, (``openzipkin/zipkin-gcp:0.13.2``) and only setting the following env variable, letting zipkin discover and use the service account:
```
        env:
        - name: STORAGE_TYPE
          value: stackdriver
```

Both in the working (staging) and non-working (production) setup, there is nothing in the zipkin logs after the startup messages.

I have tried to enable more logging by setting JAVA_OPTS:
```
        - name: JAVA_OPTS
          value: "-Dlogging.level.zipkin=DEBUG -Dlogging.level.zipkin2=DEBUG -Dlogging.level.zipkin2.translation.stackdriver=DEBUG"
```

This resulted in debug messages that print translated stackdriver spans, but no errors about them failing to be forwarded.

GET /health shows:
```
{"status":"UP","zipkin":{"status":"UP","details":{"StackdriverSender{redacted-project-id}":{"status":"UP"}}}}
```

GET /metrics shows:
```
{"counter.zipkin_collector.messages.http":277.0,"counter.zipkin_collector.spans_dropped.http":0.0,"gauge.zipkin_collector.message_bytes.http":2303.0,"counter.zipkin_collector.bytes.http":681050.0,"gauge.zipkin_collector.message_spans.http":3.0,"counter.zipkin_collector.spans.http":886.0,"counter.zipkin_collector.messages_dropped.http":0.0}
```

What would be the next step? How can I debug this further?

Thanks you in advance!

This got me the following error:

```
Error: Could not find or load main class JAVA_OPTS=-Dlogging.level.com.linecorp.armeria.logging.traffic.client=trace
Caused by: java.lang.ClassNotFoundException: JAVA_OPTS=-Dlogging.level.com.linecorp.armeria.logging.traffic.client=trace
```

--- a/README.md
+++ b/README.md
@@ -120,15 +120,14 @@ labels {
 
 If you are running Zipkin server, `/health` HTTP endpoint can be used to check service health.
 
-## gRPC Headers
+### gRPC Headers
 
 If you suspect an issue between Zipkin server and Stackdriver, inspecting gRPC headers may be useful.
 Set `STACKDRIVER_HTTP_LOGGING` environment variable to `HEADERS` to log gRPC status information.
 
 ### GCP console
 
-If you believe that spans are reaching Stackdriver, verify what happens to them from the GCP side:
-Go to APIs & Services > Dashboard > Stackdriver Trace API > Metrics.
+If you believe that spans are reaching Stackdriver, verify what happens to them from the GCP side by visiting APIs & Services > Dashboard > Stackdriver Trace API > Metrics section.
 
 * Is there any traffic at all? If not, then the requests are likely not reaching Stackdriver at all. Check your applications/proxies to make sure requests go to the right place.
 * Are there errors? If so, narrow down the source of errors by selecting specific Credentials and Methods values from the filter drop-downs on top and seeing the effect in "Errors by API method" chart or "Methods" table.

--- a/README.md
+++ b/README.md
@@ -71,7 +71,9 @@ yourself by downloading a couple jars.
 [Here's an example](autoconfigure/storage-stackdriver#quick-start) of
 integrating Stackdriver storage.
 
-## Troubleshooting translation issues
+## Troubleshooting
+
+### Translation issues
 
 When using a component that sends data to Stackdriver, if you see nothing in the console,
 try enabling DEBUG logging on the translation component. If using Spring Boot (ex normal
@@ -113,6 +115,26 @@ labels {
   value: "frontend"
 }
 ```
+
+### Healthcheck API
+
+If you are running Zipkin server, `/health` HTTP endpoint can be used to check service health.
+
+## gRPC Headers
+
+If you suspect an issue between Zipkin server and Stackdriver, inspecting gRPC headers may be useful.
+Set `STACKDRIVER_HTTP_LOGGING` environment variable to `HEADERS` to log gRPC status information.
+
+### GCP console
+
+If you believe that spans are reaching Stackdriver, verify what happens to them from the GCP side:
+Go to APIs & Services > Dashboard > Stackdriver Trace API > Metrics.
+
+* Is there any traffic at all? If not, then the requests are likely not reaching Stackdriver at all. Check your applications/proxies to make sure requests go to the right place.
+* Are there errors? If so, narrow down the source of errors by selecting specific Credentials and Methods values from the filter drop-downs on top and seeing the effect in "Errors by API method" chart or "Methods" table.
+    * If all writes are failing, check that your service account has access to `cloudtrace.agent` role or `cloudtrace.traces.patch` permission.
+    * If reads are failing, check for `cloudtrace.user` role or the specific permission (full permissions list).
+NOTE: read and write permissions are separate; only the admin role has both.
 
 ## Artifacts
 All artifacts publish to the group ID "io.zipkin.gcp". We use a common

